### PR TITLE
prompt and force quit

### PR DESCRIPTION
**Pull request by jpittis**
I'm so happy you added params to logmsg. ❤️ 

This adds `:q` and `:q!` and fixes trie to not be bugged again.

### CODE REVIEW

File: src/init.c [hunk 1]
```diff
@@ -1,7 +1,10 @@
 #include "init.h"
 #include "function.h"
 #include "colon.h"
+#include "trie.h"
 
 void initUser(void) {
-  registerColonFunction("q", &quit);
+  initTrie(&colonFunctions);
```
Comment (jpittis) [inline src/init.c]: do I even need this? am I right that `static` puts the variable in the "zero initialized section"?

File: src/process_keypress.c [hunk 2]
```diff
@@ -72,6 +69,8 @@ void editorProcessKeypress(int fd) {
     case 'l':
       editorMoveCursor(RIGHT);
       break;
+    case 'q':
```
Comment (jpittis) [inline src/process_keypress.c:72]: added this as a safety for debugging. just incase we break that colon code and want to debug

File: src/init.c (hunk 1 above)
Comment (epilk) [reply src/init.c]: Yerp! globals are always 0

**Review by epilk — APPROVED** [review]
lgtm

**Pull request merged**